lib/active_shipping/carriers/usps.rb

```diff
@@ -465,14 +468,39 @@ def build_world_shipment_request(action, origin, destination, package, options =
           xml.Container(container)
 
           xml.ShippingContents do |xml|
-            xml.ItemDetail do |xml|
-              xml.Description("Item delivery")
-              xml.Quantity(1)
-              xml.Value(package.value)
-              xml.NetPounds(0)
-              xml.NetOunces("%0.1f" % [package.ounces, 1].max)
-              xml.HSTariffNumber(0)
-              xml.CountryOfOrigin('United States')
+            raise NoPackageItemsError.new("package items are needed for shipping contents info") if package.package_items.blank?
+
+            package.package_items.each do |item|
+              xml.ItemDetail do |xml|
+                if item.description.blank?
+                  raise PackageItemError.new("package must have a description")
+                else
+                  xml.Description(item.description)
+                end
+
+                xml.Quantity(1)
+
+                if item.value.blank?
+                  raise PackageItemError.new("package must have a value")
+                else
+                  xml.Value(item.value)
+                end
+
+                xml.NetPounds(0)
+                xml.NetOunces("%0.1f" % [package.ounces, 1].max)
```

Comment:
Should be `[package.ounces, 0].max`, it should be okay for the `NetOunces` to be 0.